File: k12ms26/search/두용액.py
#용액
#https://hellya.tistory.com/32
import sys

def search(solution):
    left = 0; right = len(solution)-1
    s = solution[left]+solution[right]
    res = [abs(s),solution[left],solution[right]]
    while left<right :
        s = solution[left]+solution[right]
        if abs(s) < res[0] :
            res[0] = abs(s)
            res[1] = solution[left]
            res[2] = solution[right]
        if s > 0 : right -= 1
        elif s < 0 : left += 1
        else :
            return res
    return res

N = int(sys.stdin.readline())
sol = list(map(int, sys.stdin.readline().split()))
sol.sort()
#둘의 차이의 절대값이 가장 작은 두 수여야 한다
result = search(sol)
print("%d %d "%(result[1],result[2]))

File: k12ms26/search/두용액.js
//BOJ 2470 두 용액
const input = [];
require("readline")
.createInterface(process.stdin, process.stdout)
.on("line", (line) => {
  input.push(line);
})
.on("close", () => {
  solution();
  process.exit();
});

const solution = () => {
    const N = +input[0];
    let arr = input[1].split(' ').map(Number).sort((a, b) => a-b);

    let left = 0, right = arr.length-1;
    let now = [arr[left], arr[right]];
    while(left <= right) {
        console.log(arr[left], arr[right])
        if(Math.abs(now[0]+now[1]) > Math.abs(arr[left]+arr[right]) && left !== right) {
            now = [arr[left], arr[right]];
        }
        let add = arr[left]+arr[right];
        if(add > 0) right--;
        else left++;
    }

    console.log(`${now[0]} ${now[1]}`)
};